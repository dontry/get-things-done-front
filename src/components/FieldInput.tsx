import React, { Component } from "react";
import { observable, action } from "mobx";
import { observer } from "mobx-react";

export class FieldState {
  @observable
  public value = "";

  @action
  public onChange(value: string): void {
    this.value = value;
  }
}

@observer
export class FieldInput extends Component<{ fieldState: FieldState }> {
  public render() {
    const { fieldState } = this.props;
    return (
      <input
        type="text"
        value={fieldState.value}
        onChange={e => fieldState.onChange(e.target.value)}
      />
    );
  }
}

export default FieldInput;
