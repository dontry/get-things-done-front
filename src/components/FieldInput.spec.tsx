import React from "react";
import { render } from "react-testing-library";
import FieldInput, { FieldState } from "./FieldInput";

describe("FieldInput", () => {
  const fieldState: FieldState = {
    value: "",
    onChange: jest.fn()
  };
  it("should render", () => {
    const { container } = render(<FieldInput fieldState={fieldState} />);
    expect(container).toMatchSnapshot();
  });
});
