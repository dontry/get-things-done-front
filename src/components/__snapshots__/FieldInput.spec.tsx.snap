// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput should render 1`] = `
<div>
  <input
    type="text"
    value=""
  />
</div>
`;
